### PR TITLE
Contact Primary Affiliation Field is Mapped to More than Account Record Type Labels

### PR DESCRIPTION
**Pull request by xinnanlisf**
# Critical Changes

# Changes

# Issues Closed

# New Metadata

## Unpackaged Metadata

# Deleted Metadata

# Testing Notes

### CODE REVIEW

**Comment by xinnanlisf** [comment]
@MatthewBlanski the last piece of affiliation mappings puzzles. I tweaked the lookup model a bit to enable duplicate detection. Let me know if any feedback.

File: src/classes/LookupFieldModel.cls [hunk 1]
```diff
@@ -34,9 +34,11 @@
 public with sharing class LookupFieldModel {
     public String name; //developer name
     public String label;
+    public String key;
 
     public LookupFieldModel(String name, String label) {
         this.name = name;
         this.label = label;
+        this.key = name + label;
```
Comment (MatthewBlanski) [inline src/classes/LookupFieldModel.cls:42]: The name of a LookupFieldModel is guaranteed to be unique for a given sObjectField. Let's use that instead.

File: src/classes/AffiliationMappingsHealthCheckVMapper.cls [hunk 2]
```diff
@@ -770,4 +865,42 @@ public virtual with sharing class AffiliationMappingsHealthCheckVMapper {
...
+
+    /**********************************************************************************
+    * @description Retrieves a set of duplicate lookup field model keys in affiliation mappings
+    * @param  affiliationMappingsModelsList A list of models representing affiliation mappings in EDA Settings
+    * @param  accountLookupsOnContactFieldModelsByName A map of lookup field model by name
+    * @param  accountLookupsOnContactFieldModelsByLabel A map of lookup field model by label
+    * @return A set of duplicate lookup field model keys
+    **********************************************************************************/
+    @TestVisible
+    private Set<String> getDuplicateLookupFieldModelKeys(
```
Comment (MatthewBlanski) [inline src/classes/AffiliationMappingsHealthCheckVMapper.cls:877]: You can preserve this method and use the name of the lookup field model. When lacking an Id, a name suffices as our ultimate goal is to retrieve a unique identifier for a lookup field model for a given sObjectField, which is Relationship Type in this case.

File: src/classes/AffiliationMappingsHCVMapper_TEST.cls [hunk 3]
```diff
@@ -138,7 +138,7 @@ public with sharing class AffiliationMappingsHCVMapper_TEST {
         AffiliationMappingsHealthCheckVMapper AffiliationMappingsHealthCheckVMapper =
             AffiliationMappingsHealthCheckVMapper.getInstance();
 
-        Integer numberOfHealthChecks = 5; //TODO: Update when adding new Health Check
+        Integer numberOfHealthChecks = 6; //TODO: Update when adding new Health Check
```
Comment (MatthewBlanski) [inline src/classes/AffiliationMappingsHCVMapper_TEST.cls:141]: Let's change this over to a note.
```suggestion
        Integer numberOfHealthChecks = 6; //NOTE: Update when adding new Health Check
```

File: src/classes/AffiliationMappingsHCVMapper_TEST.cls [hunk 4]
```diff
@@ -194,7 +194,7 @@ public with sharing class AffiliationMappingsHCVMapper_TEST {
         AffiliationMappingsHealthCheckVMapper AffiliationMappingsHealthCheckVMapper =
             AffiliationMappingsHealthCheckVMapper.getInstance();
 
-        Integer numberOfHealthChecks = 5; //TODO: Update when adding new health checks
+        Integer numberOfHealthChecks = 6; //TODO: Update when adding new health checks
```
Comment (MatthewBlanski) [inline src/classes/AffiliationMappingsHCVMapper_TEST.cls:197]: Likewise here on changing this to a note.
```suggestion
        Integer numberOfHealthChecks = 6; //NOTE: Update when adding new health checks
```

**Review by MatthewBlanski — CHANGES_REQUESTED** [review]
A few minor changes to avoid some extra complexity that I don't believe we need to introduce given the final state of the code and one cleanup task prior to final commit. This is clean and elegant otherwise.

**Comment by xinnanlisf** [comment]
@MatthewBlanski thanks for the feedback. The PR should be ready again!

File: src/classes/LookupFieldModel_TEST.cls [hunk 5]
```diff
@@ -48,6 +48,6 @@ public with sharing class LookupFieldModel_TEST {
         Test.stopTest();
 
         System.assertEquals('TestName', lookupFieldModel.name, 'Name should be set.');
-        System.assertEquals('Test Label', lookupFieldModel.label, 'Label should be set.');
+		System.assertEquals('Test Label', lookupFieldModel.label, 'Label should be set.');
```
Comment (MatthewBlanski) [inline src/classes/LookupFieldModel_TEST.cls:51]: Spacing fix.
```suggestion
        System.assertEquals('Test Label', lookupFieldModel.label, 'Label should be set.');
```

**Review by MatthewBlanski — APPROVED** [review]

**Pull request merged**